src/rogue/utilities/fileio/StreamReader.cpp

```diff
@@ -165,20 +169,34 @@ void ruf::StreamReader::runThread() {
                break;
             }
 
+            // Skip next step if frame is empty
+            size -= 4;
```

Comment:
is it possible for size to not be a multiple of 4?  There's no checks for it.

Reply:
Yeah. That should be: 
if ( size <= 4 ) continue;
size -= 4;